Atomic/syscontainers.py

```diff
@@ -386,6 +386,147 @@ def _write_config_to_dest(self, destination, exports_dir, values=None):
+            new_installed_files_checksum = {}
+        else:
+            new_installed_files_checksum = RPMHostInstall.rm_add_files_to_host(options["installed_files_checksum"], exports, options["prefix"] or "/", files_template=installed_files_template, values=options["values"], rename_files=rename_files)
+
+        new_installed_files = list(new_installed_files_checksum.keys())
+
+        # Here, we pack the created attributes into a dictionary
+        rpm_install_content = {"new_installed_files" : new_installed_files,
+                               "new_installed_files_checksum" : new_installed_files_checksum,
+                               "rpm_installed" : rpm_installed,
+                               "rpm_file" : rpm_file}
+
+        return rpm_install_content
+
+    def _write_info_file(self, options, manifest, rpm_install_content, image_id, rev):
+        """
+        Based on the information collected before in '_do_checkout' function, we collect and write this
+        information in json format into a info file at the container location
+
+        :param options: a dictionary which contains user's input collectively
+        :param manifest: dictionary loaded from manifest json file
+        :param rpm_install_content: a dictionary collected after handling system_package related actions
+        :param image_id: the id of the image
+        :param rev: the refspec of the corresponding image
+        """
+        installed_files_template = SystemContainers._get_manifest_attributes(manifest, "installedFilesTemplate", [])
+        rename_files = SystemContainers._get_manifest_attributes(manifest, "renameFiles", {})
+        has_container_service = not(SystemContainers._get_manifest_attributes(manifest, "noContainerService", False))
+
+        try:
+            with open(os.path.join(options["destination"], "info"), 'w') as info_file:
+                info = {"image" : options["img"],
+                        "revision" : image_id,
+                        "ostree-commit": rev,
+                        'created' : calendar.timegm(time.gmtime()),
+                        "values" : options["values"],
+                        "has-container-service" : has_container_service,
+                        "installed-files": rpm_install_content["new_installed_files"],
+                        "installed-files-checksum": rpm_install_content["new_installed_files_checksum"],
+                        "installed-files-template": installed_files_template,
+                        "rename-installed-files" : rename_files,
+                        "rpm-installed" : rpm_install_content["rpm_installed"],
+                        "system-package" : options["system_package"],
+                        "remote" : options["remote"],
+                        "runtime" : self._get_oci_runtime()}
+                info_file.write(json.dumps(info, indent=4))
+                info_file.write("\n")
+        except (NameError, AttributeError, OSError, IOError) as e:
```

Comment:
an extra IOError case was added here, in case destination somehow got deleted during the checkout (and I shamelessly added it because it is also helpful for writing unit test). 

What's your thoughts on this? I can remove this error and the one in unit test check if you want too =).

Reply:
I think handling it is better as you've done *unless* it's handled higher in the call.